crates/router/src/core/errors/api_error_response.rs

```diff
@@ -301,9 +301,9 @@ impl actix_web::ResponseError for ApiErrorResponse {
             | Self::MissingDisputeId
             | Self::FileNotFound
             | Self::FileNotAvailable => StatusCode::BAD_REQUEST, // 400
-            Self::ReturnUrlUnavailable => StatusCode::SERVICE_UNAVAILABLE, // 503
-            Self::PaymentNotSucceeded => StatusCode::BAD_REQUEST,          // 400
-            Self::NotImplemented { .. } => StatusCode::NOT_IMPLEMENTED,    // 501
+            Self::ReturnUrlUnavailable => StatusCode::NOT_FOUND, // 404
```

Comment:
Should this be 400 (`BAD_REQUEST`) instead, since it indicates that a required piece of information was not provided (could be in a previous step), rather than a resource being not found?

Reply:
I mapped it to NotFound because it was mapped to NotFound here https://github.com/juspay/hyperswitch/blob/fc6acd04cb28f02a4f52ec77d8ae003957183ff2/crates/router/src/core/errors/api_error_response.rs#L477 

Thought it might have already been discussed and decided, hence, didn't raise the concern here. 

Also want to highlight that this error is not being used anywhere. Everywhere(in request, response, router_data, payment_attempt) return_url is optional.